src/main/scala/SemanticVersion.scala

```diff
@@ -1,93 +1,127 @@
 package compliant
 
 class SemanticVersion(val versionString: String) {
+  // Perhaps replace some of this with proper RegExs?
+  // I'm not sure which is more performant
   val trimmedVersionString = versionString.trim.replaceAll("\\s", "")
   val splitByPeriods = trimmedVersionString.split("\\.")
   val major = splitByPeriods(0)
   val minor = splitByPeriods(1)
   val patch = splitByPeriods.slice(2,splitByPeriods.length).mkString(".").split("\\+|\\-")(0)
 
+  // standard attributes
   def fullVersionString: String = {
     this.trimmedVersionString
   }
 
+
   def majorVersion : String = {
     this.major
   }
 
+  //
   def minorVersion : String = {
     this.minor
   }
 
+
   def patchVersion : String = {
     this.patch
   }
 
+
+  // slightly less straightforward attributes
   def versionStringWithoutBuildMetadata : String = {
     this.trimmedVersionString.split("\\+").head
   }
 
+
   def prerelease : String = {
     val splitByDashes = this.trimmedVersionString.split("\\-")
     splitByDashes.slice(1, splitByDashes.length).mkString("-").split("\\+").head
   }
 
+
+  //
   def buildMetadata : String = {
     val splitByPlus = this.trimmedVersionString.split("\\+")
     // join with plus to recreate original string
     splitByPlus.slice(1,splitByPlus.length).mkString("+")
   }
 
+
+  // comparators
   def ==(otherVersionString: String) : Boolean = {
     val otherVersion = new SemanticVersion(otherVersionString)
     this == otherVersion
   }
 
+
   def ==(otherVersion: SemanticVersion) : Boolean = {
     this.versionStringWithoutBuildMetadata == otherVersion.versionStringWithoutBuildMetadata
   }
 
+
+  //
   def >=(otherVersion: SemanticVersion) : Boolean = {
     val greaterThan = this > otherVersion
     val equalTo = this == otherVersion
     greaterThan || equalTo
   }
 
+
+  // The heart of comparisons is based off of this
+  // Suggestions for improvement welcome
```

Comment:
I think a clear way would be use a for-loop with an array [major, minor, patch, prerelease] values and then:
```
isGreater = false
for (i=0; i < arr.length ; i++1) {
  if (v1.arr[i] > v2.arr[i]) {
    isGreater = true
    break;
  } else if (v1.arr[i] < v2.arr[i]) {
    break
  }
}
```